src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageBus.cs

```diff
@@ -355,23 +367,44 @@ private void ClosePublisherConnection()
             if (_logger.IsEnabled(LogLevel.Trace))
                 _logger.LogTrace("ClosePublisherConnection");
 
-            if (_publisherModel != null)
+            if (_publisherChannel != null)
             {
-                _publisherModel.Close();
-                _publisherModel.Abort();
-                _publisherModel.Dispose();
-                _publisherModel = null;
+                _publisherChannel.Dispose();
```

Comment:
I implemented this twice due to a lot less async/sync mix. Their dispose under the hood is now doing GetAwaiter().GetResult() but only if it's open, the rest is sync.